src/mlpack/tests/decision_tree_regressor_test.cpp

```diff
@@ -0,0 +1,1047 @@
+  arma::rowvec predictions;
+  tree.Predict(testData, predictions);
+
+  REQUIRE(predictions.n_elem == testData.n_cols);
+
+  // Make sure we get reasonable rmse.
+  const double rmse = RMSE(predictions, testLabels);
+  REQUIRE(rmse < 1.0);
+}
+
+/**
+ * Test that we can build a decision tree with weights on a simple categorical
+ * dataset.
+ */
+TEST_CASE("CategoricalBuildTestWithWeight_", "[DecisionTreeRegressorTest]")
+{
+  arma::mat d;
+  arma::rowvec l;
+  data::DatasetInfo di;
+  MockCategoricalData(d, l, di);
+
+  // Split into a training set and a test set.
+  arma::mat trainingData = d.cols(0, 1999);
+  arma::mat testData = d.cols(2000, 3999);
+  arma::rowvec trainingLabels = l.subvec(0, 1999);
+  arma::rowvec testLabels = l.subvec(2000, 3999);
+
+  arma::rowvec weights = arma::ones<arma::rowvec>(
+      trainingLabels.n_elem);
+
+  // Build the tree.
+  DecisionTreeRegressor<> tree(trainingData, di, trainingLabels, weights, 10);
+
+  // Now evaluate the quality of predictions.
+  arma::rowvec predictions;
+  tree.Predict(testData, predictions);
+
+  REQUIRE(predictions.n_elem == testData.n_cols);
+
+  // Make sure we get reasonable rmse.
+  const double rmse = RMSE(predictions, testLabels);
+  REQUIRE(rmse < 1.0);
+}
+
+/**
+ * Test that we can build a decision tree using weighted data (where the
+ * low-weighted data is random noise), and that the tree still builds correctly
+ * enough to get good results.
+ */
+// TEST_CASE("WeightedDecisionTreeTest_", "[DecisionTreeRegressorTest]")
+// {
+//   // Loading data.
+//   data::DatasetInfo info;
+//   arma::mat trainData, testData;
+//   arma::Row<double> trainLabels, testLabels;
+//   LoadBostonHousingDataset(trainData, testData, trainLabels, testLabels, info);
+
+//   // Add some noise.
+//   arma::mat noise(trainData.n_rows, 500, arma::fill::randu);
+//   arma::Row<double> noiseLabels(500);
+//   for (size_t i = 0; i < noiseLabels.n_elem; ++i)
+//     noiseLabels[i] = 15 + math::Random(0, 10); // Random label.
+
+//   // Concatenate data matrices.
+//   arma::mat data = arma::join_rows(trainData, noise);
+//   arma::Row<double> fullLabels = arma::join_rows(trainLabels, noiseLabels);
+
+//   // Now set weights.
+//   arma::rowvec weights(trainData.n_cols + 500);
+//   for (size_t i = 0; i < trainData.n_cols; ++i)
+//     weights[i] = math::Random(0.9, 1.0);
+//   for (size_t i = trainData.n_cols; i < trainData.n_cols + 500; ++i)
+//     weights[i] = math::Random(0.0, 0.01); // Low weights for false points.
+
+//   // Now build the decision tree.  I think the syntax is right here.
+//   DecisionTreeRegressor<> d(data, fullLabels, weights);
+
+//   // Now we can check that we get good performance on the VC2 test set.
+//   arma::Row<double> predictions;
+//   d.Predict(testData, predictions);
+
+//   REQUIRE(predictions.n_elem == testData.n_cols);
+
+//   // Figure out the accuracy.
+//   double rmse = RMSE(predictions, testLabels);
+
+//   REQUIRE(rmse < 9.21);
+// }
```

Comment:
Do you want to uncomment this test now?  Or remove the Boston housing dataset?  Up to you.

Reply:
We can't remove the Boston housing dataset. There are some tests like `SimpleGeneralizationTest`, this one and a couple more that requires a real-world dataset. So, I would say we should keep this for now. Once DatasetInfo is fixed, I will complete these tests.